pkg/validate/validate.go

```diff
@@ -113,3 +113,12 @@ func URI(uri string) error {
 	}
 	return nil
 }
+
+// URIWithProtocol validates that the URI specifies a certain
+// protocol scheme (e.g. "https://") before validating the URI
+func URIWithProtocol(uri string, protocol string) error {
+	if !strings.HasPrefix(strings.ToLower(uri), protocol) {
```

Comment:
you can parse to URL and check for scheme.... ? rather than string prefix.

Reply:
Updated.